pangeo_forge_recipes/recipes/xarray_zarr.py

```diff
@@ -390,14 +392,19 @@ def calculate_sequence_lens(
     all_lens = np.array([m["dims"][concat_dim] for m in input_meta.values()])
     all_lens.shape = list(file_pattern.dims.values())
 
-    # check that all lens are the same along the concat dim
-    concat_dim_axis = list(file_pattern.dims).index(concat_dim)
-    selector = [slice(0, 1)] * len(file_pattern.dims)
-    selector[concat_dim_axis] = slice(None)  # this should broadcast correctly agains all_lens
-    sequence_lens = all_lens[tuple(selector)]
-    if not (all_lens == sequence_lens).all():
-        raise ValueError(f"Inconsistent sequence lengths found: f{all_lens}")
-    return np.atleast_1d(sequence_lens.squeeze()).tolist()
+    if len(all_lens.shape) == 1:
+        all_lens = all_lens.reshape((all_lens.shape[0], 1))
+
+    unique_vec, unique_idx = np.unique(all_lens, axis=1, return_index=True)
+    if len(unique_idx) > 1:
+        err_idx = np.where(scipy.stats.mode(all_lens, axis=1)[0] != all_lens)
```

Comment:
Can we think of a clever way to do this without resorting to scipy.stats.mode? What does this line actually do?

Reply:
I am trying to calculate the most frequent value in each row -- I am pretty sure I an re-create this with pure numpy.

Reply:
I want to find the "typical" values in order to find the places where there are discrepancies.

Reply:
`np.bincount` (i.e., histogram with width 1)

Reply:
Just realized: `unique` returns sorted unique vectors -- so, I probably don't need any other functions in the first place. Patching now...